### PR TITLE
Handle parenthesised expression sequences & add full support for PostfixExpression CST node transformation

### PR DESCRIPTION
**Pull request by ianyong**
Renamed `ExpressionSequence` to `SequenceExpression` to better reflect that a sequence of expressions is itself an expression. This is necessary because the following is a valid C program that returns 6 followed by 9:
```c
#include <stdio.h>

int main() {
    int i = 3;
    printf("%d\n", (7 - 2, i *= 3, 7 - 4) * 2);
    printf("%d\n", i);
}
```

### CODE REVIEW

File: src/interpreter/explicitControlEvaluator.ts [hunk 1]
```diff
@@ -162,6 +170,10 @@ const evaluators: AgendaItemEvaluatorMapping = {
     command: EmptyStatement,
     state: ExplicitControlEvaluatorState
   ) => {},
+  SequenceExpression: (
```
Comment (zognin) [inline src/interpreter/explicitControlEvaluator.ts]: Shall we move this down to be in lexicographical order?

File: src/ast/astBuilder.ts [hunk 2]
```diff
@@ -999,33 +998,97 @@ export class ASTBuilder implements CVisitor<any> {
...
+          const arrayAccessIndexExpression = ctx.expression(
+            currArrayAccessIndexExpressionIdx++
+          );
+          const indexBeingAccessed = this.visitExpression(
+            arrayAccessIndexExpression
+          );
+          expression = {
+            type: 'ArrayAccessExpression',
+            expression,
+            indexBeingAccessed
```
Comment (zognin) [inline src/ast/astBuilder.ts:1012]: Will this always be the index being accessed? What about declarations like `int arr[5];` is 5 the index being accessed? I thought it's the size. Edit: Oh I see that that's handled separately under declarations, nvm then.

**Review by zognin — APPROVED** [review]
LGTM I left 1 question

File: src/interpreter/explicitControlEvaluator.ts (hunk 1 above)
Comment (ianyong) [reply src/interpreter/explicitControlEvaluator.ts]: Oh yes, thanks for the catch!

**Pull request merged**